### PR TITLE
Lib-prefixed to multioutput: ffado

### PR DESCRIPTION
**Pull request by avnik**
###### Motivation for this change

Say "farewell" to long-lived ad-hoc hack, when we had two derivations for one package -- one for library-only version, second one with mixer built.

This PR contain finalized ffado rework, factored out of #38494
(because it hard to maintain so long live branch, which cause significant local rebuild)

ffado itself splitted into two derivations: `ffado` and `ffado-mixer` (latter contain some qt/dbus based mixer). This was needed to avoid build dependency loop.

See #14693 for problem history and details.

PS. Would be nice, if someone who have actual ffado backed hardware to test

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
Only shotcut fails to build, but this is currently broken on hydra. cc @woffs

File: pkgs/os-specific/linux/ffado/default.nix [hunk 1]
```diff
@@ -62,25 +73,15 @@ stdenv.mkDerivation rec {
...
   '';
 
-  installPhase = if libOnly then ''
-    scons PREFIX=$TMPDIR UDEVDIR=$TMPDIR \
-      LIBDIR=$out/lib INCLUDEDIR=$out/include install
-  '' else ''
-    scons PREFIX=$out PYPKGDIR=$PYDIR UDEVDIR=$out/lib/udev/rules.d install
-  '' + stdenv.lib.optionalString (optPyqt4 != null && optPythonDBus != null) ''
+  installPhase = ''
+    scons ${SCONS_OPTIONS} ${if mixerOnly then "${BINDIR} ${PYDIR}" else "install"}
```
Comment (matthewbauer) [inline pkgs/os-specific/linux/ffado/default.nix]: You don't have to do this here, but eventually I want to get scons to have setup hook that defines these phases (like is done with cmake, ninja, etc...).

Comment (avnik) [reply pkgs/os-specific/linux/ffado/default.nix]: IDK other legal way to bypass library build, when we build only mixer. `scons $out/lib/python $out/bin` force installing only needed stuff.

Comment (avnik) [reply pkgs/os-specific/linux/ffado/default.nix]: @matthewbauer what can I do, to get this one PR merged? Is a deepest rebuild trigger in my WIP tree at the moment.  (probably we can even drop mixer temporary, and make separate package for it, if anyone need it really)

**Comment by matthewbauer** [comment]
Sorry for the lateness. Merged in 78cabc3ca9edfd04b93230aa74c132a0f81a2a25.

**Pull request closed without merging**